crates/fuzzing/src/oracles.rs

```diff
@@ -479,3 +476,189 @@ pub fn table_ops(config: crate::generators::Config, ops: crate::generators::tabl
         }
     }
 }
+
+/// Do we accept the given wasm module for differential fuzzing between Wasmtime and wasmi? We want
+/// to return early for modules that don't meet our criteria so that the fuzzer learns that these
+/// are not interesting (as they will not lead to additional coverage).
+///
+/// If accepted, returns the name of the memory we will diff and the function we will call.
+fn differential_wasmi_accept_module(wasm: &[u8]) -> Option<(String, String)> {
```

Comment:
This function may actually be best done elsewhere perhaps? It might make more sense to create a `wasmtime::Config` and use wasmtime to compile the module first, and that can do a lot of the gating here. We can disable wasm features that wasmi hasn't implemented to prevent them getting past validation.

Additionally once we've got `wasmtime::Module` we can inspect the exports for what to locate.

Reply:
Another alternative is to configure `wasm_smith::Config` to generate only appropriate modules for running in wasmi. We could presumably add configuration options for things like "disable this proposal", "memories can be at most this big", or "there must be a `main` function which takes no arguments"

Reply:
Done!